datasetUtils/checkForNewVersion.go

```diff
@@ -49,19 +62,19 @@ func CheckForNewVersion(client *http.Client, APP string, VERSION string, interac
 		log.Fatalf("Illegal version number:%v", VERSION)
 	}
 	log.Printf("Latest version: %s", latestVersion)
-	if majorLatest > majorCurrent {
-		log.Println("You must upgrade to a newer version")
-		log.Println("Current Version: ", VERSION)
-		log.Println("Latest  Version: ", latestVersion)
-		log.Println("Use the following command to update:")
-		log.Println("curl -O " + DeployLocation + APP + ";chmod +x " + APP)
-		log.Fatalf("Execution stopped, please update program now.\n")
-	} else if version.Compare(latestVersion, VERSION, ">") {
+	
+	// Get the operating system name
+	osName := runtime.GOOS
+	
+	// Generate the download URL
+	downloadURL := fmt.Sprintf("%s/scicat-cli_%s_%s_x86_64.tar.gz", DeployLocation, latestVersion, strings.Title(osName))
+	
+	if majorLatest > majorCurrent || version.Compare(latestVersion, VERSION, ">") {
 		log.Println("You should upgrade to a newer version")
 		log.Println("Current Version: ", VERSION)
 		log.Println("Latest  Version: ", latestVersion)
 		log.Println("Use the following command to update:")
-		log.Println("curl -O " + DeployLocation + APP + ";chmod +x " + APP)
+		log.Printf("curl -L -O %s; tar xzf scicat-cli_.%s_%s_x86_64.tar.gz; chmod +x %s\n", downloadURL, latestVersion, strings.Title(osName), APP)
```

Comment:
should we say to use WSL for windows?

Reply:
Windows doesn't need chmod. The rest of the command is now different for windows.